src/entities/mod.rs

```diff
@@ -0,0 +1,9 @@
+use crate::error::DomainError;
+
+pub(crate) mod record;
+
+pub(crate) trait TryToEntity {
+    type OutputEntity;
+
+    fn try_to_entity(self) -> Result<Self::OutputEntity, DomainError>;
+}
```

Comment:
From what I see, this trait is pretty much [`try_from`](https://doc.rust-lang.org/stable/std/convert/trait.TryFrom.html#), I don’t think you need it.